File: app/src/test/java/com/mgaetan89/showsrage/fragment/AddShowFragment_IsQueryValidTest.java
package com.mgaetan89.showsrage.fragment;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.Parameterized;

import java.util.Arrays;
import java.util.Collection;

import static org.assertj.core.api.Assertions.assertThat;

@RunWith(Parameterized.class)
public class AddShowFragment_IsQueryValidTest {
	@Parameterized.Parameter(0)
	public String query;

	@Parameterized.Parameter(1)
	public boolean valid;

	@Test
	public void isQueryValid() {
		assertThat(AddShowFragment.Companion.isQueryValid(this.query)).isEqualTo(this.valid);
	}

	@Parameterized.Parameters
	public static Collection<Object[]> data() {
		return Arrays.asList(new Object[][]{
				{null, false},
				{"", false},
				{" ", false},
				{"  ", false},
				{" some query ", true},
		});
	}
}

File: app/src/test/kotlin/com/mgaetan89/showsrage/fragment/AddShowFragment_IsQueryValidTest.kt
package com.mgaetan89.showsrage.fragment

import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.Parameterized

@RunWith(Parameterized::class)
class AddShowFragment_IsQueryValidTest(val query: String?, val valid: Boolean) {
	@Test
	fun isQueryValid() {
		assertThat(AddShowFragment.isQueryValid(this.query)).isEqualTo(this.valid)
	}

	companion object {
		@JvmStatic
		@Parameterized.Parameters
		fun data(): Collection<Array<Any?>> {
			return listOf(
					arrayOf<Any?>(null, false),
					arrayOf<Any?>("", false),
					arrayOf<Any?>(" ", false),
					arrayOf<Any?>("  ", false),
					arrayOf<Any?>(" some query ", true)
			)
		}
	}
}
